### PR TITLE
oobmigrations: Add metrics

### PR DESCRIPTION
**Pull request by efritz**
Fixes https://github.com/sourcegraph/sourcegraph/issues/18406.

### CODE REVIEW

File: doc/admin/observability/dashboards.md [hunk 1]
```diff
@@ -516,6 +516,44 @@ This panel indicates index enqueuer errors every 5m.
...
 
+### Frontend: Out of band migrations
+
+#### frontend: out_of_band_migrations_up_99th_percentile_duration
+
+This panel indicates 99th percentile successful out-of-band up migration operations duration over 5m.
+
+<sub>*Managed by the [Sourcegraph Core application team](https://about.sourcegraph.com/handbook/engineering/core-application).*</sub>
+
+<br />
```
Comment (Strum355) [inline doc/admin/observability/dashboards.md:527]: Is this meant to be the time for an _entire_ migration or individual "batches" of a migration? If the former, does it make sense to have it given that itll probably be 0 for like 90% of the time lol

Comment (efritz) [reply doc/admin/observability/dashboards.md:527]: Each batch.

Comment (efritz) [reply doc/admin/observability/dashboards.md:527]: Feel free to suggest a more intuitive name if you've got one :)

Comment (Strum355) [reply doc/admin/observability/dashboards.md:527]: imo for these names that have underscores, `oob` would be fine and give us more "character budget" to make things more descriptive:
`out_of_band_migrations_up_99th_percentile_duration` ->
`oob_migrations_up_per_op_99th_percentile_duration` for example

Comment (Strum355) [reply doc/admin/observability/dashboards.md:527]: I shouldve voiced my thoughts about using the name "batch" given we have Batch Changes as a product :sweat_smile: naming things eh :facepalm:

Comment (efritz) [reply doc/admin/observability/dashboards.md:527]: I'm not going to stop using common words with a very precise meaning because we keep rebranding 🤣

Comment (efritz) [reply doc/admin/observability/dashboards.md:527]: Clarified.

**Review by Strum355 — APPROVED** [review]

File: internal/oobmigration/observability.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		observationContext.Registerer,
+		"oobmigration",
+		metrics.WithLabels("op", "migration"),
+		metrics.WithCountHelp("Total number of migrator invocations."),
+	)
+
+	opForMigration := func(name string) func(migrationID int) *observation.Operation {
+		return func(migrationID int) *observation.Operation {
+			return observationContext.Operation(observation.Op{
+				Name:         fmt.Sprintf("oobmigration.%s", name),
```
Comment (eseliger) [inline internal/oobmigration/observability.go:27]: is it a problem that name can have spaces?

**Review by eseliger — APPROVED** [review]

File: internal/oobmigration/observability.go (hunk 2 above)
Comment (efritz) [reply internal/oobmigration/observability.go:27]: They don't by convention (it's just method names).

**Pull request merged**